common/models/middleware.js

```diff
@@ -16,6 +16,23 @@ module.exports = function(Middleware) {
     }
     return phase;
   };
+  Middleware.getPhaseFromId = function(id) {
+    const parts = id.split('.');
+    if (parts.length && parts.length > 1) {
+      let phase = parts[0];
```

Comment:
22-26 could be rewritten as:

`return parts[0] + (parts.length > 2 ? ':' + parts[1] : '');`